### PR TITLE
Manage reporter data

### PR DESCRIPTION
**Pull request by camlafit**
A simple proposition to link reporter to each issue

Reporter are also related to other context as note issue. It's why is set as an object and not array.

### CODE REVIEW

File: src/Resources/Reporter.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+     * @var string
+     */
+    public $email;
+
+    /**
+     * Create a new resource instance.
+     *
+     * @param $attributes
+     */
+    public function __construct(array $attributes)
```
Comment (stefanius) [inline src/Resources/Reporter.php:40]: Now all the attributes are optional? At least what a reporter has is a id I asume? And very likely you get the name and mailladdress also?

https://documenter.getpostman.com/view/29959/mantis-bug-tracker-rest-api/7Lt6zkP#13f890da-50ff-0ab3-ff66-32b701d90c66

File: tests/IssuesTest.php [hunk 2]
```diff
@@ -173,6 +174,7 @@ public function it_should_create_an_issue()
                 'summary' => $this->issue['summary'],
                 'description' => $this->issue['description'],
                 'category' => $this->issue['category']['name'],
+                'reporter' => $this->issue['reporter']->toArray(),
```
Comment (stefanius) [inline tests/IssuesTest.php:177]: Maybe it is better to use the same approach as we did on category?

File: tests/IssuesTest.php [hunk 3]
```diff
@@ -29,6 +29,7 @@ protected function setUp(): void
             'description' => 'Description',
             'category' => ['id' => '1', 'name' => 'Category'],
             'project' => $this->project,
+            'reporter' => new \TestMonitor\Mantis\Resources\Reporter(['id' => '1', 'name' => 'Reporter', 'real_name' => 'My real name', 'email' => 'reporter@example.tld']),
```
Comment (stefanius) [inline tests/IssuesTest.php:32]: Can you do in in an array format? Just like category?

**Comment by stefanius** [comment]
Hi @camlafit,

Thank you (again) for your improvement. I just reviewed it.

File: src/Resources/Reporter.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace TestMonitor\Mantis\Resources;
+
+class Reporter extends Resource
```
Comment (stefanius) [inline src/Resources/Reporter.php:5]: Could you rename this to User? Maybe you also want to use the assignee somewhere and it are all user objects.

Comment (camlafit) [reply src/Resources/Reporter.php:5]: In api Reporter name is used, I can rename it to User but looks differ to api namming

**Comment by camlafit** [comment]
Hello

Thanks for your review, I'll check all these days and  do patches required.

**Comment by stefanius** [comment]
@camlafit Is this PR already done for a review? Of can it be closed?

**Comment by camlafit** [comment]
Hello

I was under work. I can do a check this week or next week. Then I could confirm its state.

Sorry for the delay :/

**Comment by stefanius** [comment]
> Hello
> 
> I was under work. I can do a check this week or next week. Then I could confirm its state.
> 
> Sorry for the delay :/

No problem!

**Comment by stefanius** [comment]
@camlafit This PR is open for more than a year. I will close this for now, if you still want to contribute to our Mantis client, feel free to open a new PR.

**Pull request closed without merging**